packages/backend/src/outgoing_payment/service.ts

```diff
@@ -0,0 +1,147 @@
+import { TransactionOrKnex } from 'objection'
+import * as Pay from '@interledger/pay'
+import { RatesService } from 'rates'
+import { BaseService } from '../shared/baseService'
+import { OutgoingPayment, PaymentIntent, PaymentState } from './model'
+import { AccountService } from '../account/service'
+import { ConnectorService } from '../connector/service'
+import { PaymentProgressService } from '../payment_progress/service'
+import { IlpPlugin } from './ilp_plugin'
+import * as lifecycle from './lifecycle'
+import * as worker from './worker'
+
+export interface OutgoingPaymentService {
+  get(id: string): Promise<OutgoingPayment | undefined>
+  create(options: CreateOutgoingPaymentOptions): Promise<OutgoingPayment>
+  approve(id: string): Promise<OutgoingPayment>
+  cancel(id: string): Promise<OutgoingPayment>
+  requote(id: string): Promise<OutgoingPayment>
+  processNext(): Promise<string | undefined>
+}
+
+export interface ServiceDependencies extends BaseService {
+  knex: TransactionOrKnex
+  slippage: number
+  quoteLifespan: number // milliseconds
+  accountService: AccountService
+  connectorService: ConnectorService
+  ratesService: RatesService
+  makeIlpPlugin: (sourceAccountId: string) => IlpPlugin
+  paymentProgressService: PaymentProgressService
+}
+
+export async function createOutgoingPaymentService(
+  deps_: ServiceDependencies
+): Promise<OutgoingPaymentService> {
+  const deps = {
+    ...deps_,
+    logger: deps_.logger.child({ service: 'OutgoingPaymentService' })
+  }
+  return {
+    get: (id) => getOutgoingPayment(deps, id),
+    create: (options: CreateOutgoingPaymentOptions) =>
+      createOutgoingPayment(deps, options),
+    approve: (id) => approvePayment(deps, id),
+    cancel: (id) => cancelPayment(deps, id),
+    requote: (id) => requotePayment(deps, id),
+    processNext: () => worker.processPendingPayment(deps)
+  }
+}
+
+async function getOutgoingPayment(
+  deps: ServiceDependencies,
+  id: string
+): Promise<OutgoingPayment | undefined> {
+  return OutgoingPayment.query(deps.knex)
+    .findById(id)
+    .withGraphJoined('outcome')
+}
+
+type CreateOutgoingPaymentOptions = PaymentIntent & { superAccountId: string }
+
+async function createOutgoingPayment(
```

Comment:
We should add a TODO/Issue to make this idempotent to prevent accidentally creating two outgoing payments

Reply:
Good idea. For `autoApprove: false` payments (which I assume will be the default) it's not a big issue, since one of the payments would just not get approved.

I'm not sure what the use case for `autoApprove:true` is, but in that case you're right, it should be idempotent.